docs/new-feature-tutorial.md

```diff
@@ -9,7 +9,11 @@ data system in real-time to all browsers the user may have open.
 As you read this, you may find you need to learn about Zulip's
 real-time push system; the
 [real-time push and events](events-system.html) documentation has a
-detailed explanation of how everything works.
+detailed explanation of how everything works. You may also find it
+beneficial to read Zulip's [architectural overview](architecture-overview.html).
+Zulip is a web application built using the Django framework, and some of the
```

Comment:
We should link to Django here, and maybe also link to the directory structure article (noting that it has links to Django's docs on views/models/etc).